sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -136,6 +134,8 @@ abstract class Optimizer(sessionCatalog: SessionCatalog)
     Batch("LocalRelation", fixedPoint,
       ConvertToLocalRelation,
       PropagateEmptyRelation) ::
+    Batch("Check Cartesian Products", Once,
```

Comment:
we should also add a comment here about the positioning ...

Reply:
Make sense. The PR is merged, what should I do now..

Reply:
Create a follow-up PR

Reply:
@gatorsmile I see, thanks!